File: 125-valid-palindrome/valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        StringBuilder onlyLetters = new StringBuilder();
        int n = s.length();
        for (int i = 0; i < n; i++) {
            if (Character.isLetterOrDigit(s.charAt(i))) {
                onlyLetters.append(Character.toLowerCase(s.charAt(i)));
            }
        }
        return new String(onlyLetters).equals(new String(onlyLetters.reverse()));
        
    }
}

File: 125-valid-palindrome/valid-palindrome.kt
class Solution {
    fun isPalindrome(s: String): Boolean {
        val onlyLetters: StringBuilder = StringBuilder()
        val n: Int = s.length
        for (i in 0..n - 1) {
            if (s[i].isLetterOrDigit()) {
                onlyLetters.append(s[i].toLowerCase())
            }
        }
        return onlyLetters.contentEquals(onlyLetters.reversed())
    }
}